src/Graphing.hs

```diff
@@ -37,6 +39,7 @@ import Data.Maybe (catMaybes)
 import Data.Set (Set)
 import Data.Set qualified as S
 import Prelude hiding (filter)
+import Prelude qualified as P
```

Comment:
I've addressed this in commit now. Since filter is defined in the file already, I can't import. But doing `import Prelude qualified as P (filter)`. That should avoid whole lots of default imports now.